### ISSUE TITLE
Lookup Fields still not working On Premises

### ISSUE THREAD

**Issue by NPrice99** [SharePoint/PnP-Sites-Core#229]
Hi
It looks as if Lookup fields are still not working.....

Get-SPOProvisioningTemplate creates the template xml file and for a lookup field it looks like this.

```
    <Field Type="Lookup" DisplayName="Branch" Required="FALSE" EnforceUniqueValues="FALSE" WebId="0cbbbcee-47b7-4109-abb9-c73ec7157ef0" ShowField="Title" UnlimitedLengthInDocumentLibrary="FALSE" Group="DH Columns" ID="{904ffc10-12cb-4b57-8311-0ed34e8cc447}" SourceID="0cbbbcee-47b7-4109-abb9-c73ec7157ef0" StaticName="Branch" Name="Branch" List="{{listid:Branch}}" AllowDeletion="TRUE" />
```

When I run Apply-SPOProvisioningTemplate the field is created so (SPManager output)

[spmanagerevidence.doc.docx](https://github.com/OfficeDev/PnP-Sites-Core/files/51974/spmanagerevidence.doc.docx)

This results in the field being shown so 

![branch site columns](https://cloud.githubusercontent.com/assets/8484048/11588931/289786e8-9a7c-11e5-8240-50cf18fe4038.png)

As can be seen Lookup fields still do not transfer correctly from A -> B

This script fixes the issue -

Function RepairSiteLookupColumns($webURL, $columnName, $lookupListName, $lookupWebURL)
{
# Get web, list and column objects

$web = Get-SPWeb $webURL
$lookupWeb = Get-SPWeb $lookupWebURL
$column = $web.Fields[$columnName]
$lookupList = $web.Lists[$lookupListName]
$newLookupListID = "{"+$lookupList.ID.ToString()+"}"
$newLookupWebID = $web.ID.ToString()

$column.SchemaXml
# Change schema XML on the lookup column

$column.SchemaXml = $column.SchemaXml.Replace($column.LookupWebId.ToString(), $newLookupWebID)
if ($column.LookupList)
{
    $column.SchemaXml = $column.SchemaXml.Replace($column.LookupList.ToString(), $newLookupListID)

}
else
{
    Write-Host -ForeGroundColor Yellow "LookupList is null"
    $column.LookupList = $newLookupListId
}
$column.Update()
$column.SchemaXml
# Write confirmation to console and dispose of web object

write-host "In Site" $web.Url "column " $column.Title "in Liste" $list.Title "is connected with"  $lookupList.Title "from" $lookupWeb.Url
$web.Dispose()
$lookupWeb.Dispose()
}

$webURL = "https://thesite/sites/lookup"
$listName = "Branch"
$columnName = "Branch"
$lookupListName = "Branch"
$lookupWebURL = "https://thesite/sites/lookup"

RepairSiteLookupColumns -webURL $webURL -columnName $columnName -lookupListName $lookupListName -lookupWeb $lookupWebURL

Regards

Nigel

**Comment by SuleymanA** [comment]
I am having the same problem in SharePoint online. I have also looked at other site columns created with the provisioning engine and they also have a incorrect sourceID. They are created with the source id of the template site... Thx for the powershell script. I will attempt to do something similar with CSOM.

I have had other issues but this is really a show stopper for using the provisioning framework. I will advise my colleagues and customers not to use it as this kind of essential functionality is failing and resulting in broken columns, content types and basically a broken site..

**Comment by tberst** [comment]
I tested lookup provisioning on SharePoint online yesterday and managed to get it to work as expected by replacing the 'List' value in the template.xml. Once I put the list URL : 
*List="/lists/LIST_URL"*
, the lookup column is provisioned correctly

**Comment by NPrice99** [comment]
Hi Thomas

Thanks for this.

It works Ok on SharePoint Online.

The issue is with SharePoint On-Premises.

I am back to work on Monday and I will fix it then.  (The { before the web id and list id and the } afterwards are missing.)

Regards

Nigel

**Comment by PaoloPia** [comment]
Hi Nigel,
As far as I can see this issue has been fixed by Erwin in #580 . Thus, I will close this item. Thanks for your feedback.
